### DISCUSSION TITLE
淺談 AI 工程師

### DISCUSSION THREAD

**Discussion by BeaGle0928** [1101-GS4719-C/Python-Community-GS4719-C#1651] [General]
我從小就時常聽到工程師這個職業，而在台灣這個職業也可能是很多人夢寐以求的工作，而隨著 Artificial Intelligence 逐年進步 AI 工程師這個職業也逐漸映入眾人眼簾。那到底什麼是 AI 工程師呢？
   簡單來說，不論是將AI導入產品亦或是產品即 AI 本身，AI 工程師主要負責回答三個問題，「我們能做到什麼？」、「要怎麼做到？」、「要怎麼上線及優化？」；不同於 ML 工程師， AI 工程師涵蓋的範圍更廣，這可以延伸到機器學習與人工智慧本身。以我自己的認知來說，AI 人工智慧傾向於模仿人的思考、行為等模式，而 ML 機器學習更像是教導機器去學習其工作模式，擺脫人為的控制，達成自動智慧化控制的目的，而機器學習是人工智慧下的其中一個子集。
   身為一名 AI 工程師，你我該具備怎樣的能力呢？以下列舉四點；
一、對於所有開、閉源模型都非常熟悉，能夠自由運用、並知道其限制。
二、能夠掌握最新的 AI 發展。
三、熟悉最新的 AI 工具，如 LangChain、向量資料庫及 Hugging Face 等。
   這邊簡介一下 LangChain，LangChain 是一個開發由語言模型 (Language Models) 驅動的程式框架，它提供了一個簡單的方式讓開發者能夠連接這些不同的語言模型，甚至可以使用Hugging Face平台上的各種開源模型，這就是LangChain名字的由來—Lang代表語言模型，Chain代表這些模型的連接。

四、能夠掌握 AI 模型上線以及測試的各種細節，包括 Model Drift, Scaling, Latency, Data Privacy 等。
   我認為 AI 在未來的發展無可限量，同時 AI 工程師的人均產值也相當可觀，但要成為一名 AI 工程師也必須時時刻刻精進自己，不斷汲取新知，儘管在這個資訊發達的時代這並不困難，但要時刻保持好奇心以及上進心才能保持競爭力。才是有了以上的簡介，不知道有沒有幫助你更認識這個職業或甚至對其產生興趣呢？

**Comment by 1chooo** [comment]
這些內容我在暑期實習的時候都有接觸到，尤其 LangChain 在暑假基本把官方文件都翻遍了！畢竟沒有這些框架，所有相關的工程內容都需要自己刻那根本時間會不夠，不過也因此看到像 Microsoft, Nvdia 這種科技巨頭的超強實習生團隊自己手刻出的專案，真是佩服！（Nvdia 的實習團隊幾乎都是 Standford, MIT, Cal Tech 這種等級的，太猛啦！）
